integration_test/sql/migration.exs

```diff
@@ -101,6 +101,75 @@ defmodule Ecto.Integration.MigrationTest do
     end
   end
 
+  defmodule MigrationTestTable do
+    use Ecto.Model
+
+    schema "migration_test_table" do
+      field :to_be_modified, :string
+      field :to_be_removed, :integer
+    end
+  end
+
+  defmodule AlterColumnMigration do
+    use Ecto.Migration
+
+    def up do
+      create table(:migration_test_table) do
+        add :to_be_modified, :integer
+        add :to_be_removed, :integer
+      end
+
+      alter table(:migration_test_table) do
+        modify :to_be_modified, :string
+      end
+
+      execute "INSERT INTO migration_test_table (to_be_modified) VALUES ('foo')"
+    end
+
+    def down do
+      drop table(:migration_test_table)
+    end
+  end
+
+  defmodule DropColumnMigration do
+    use Ecto.Migration
+
+    def up do
+      create table(:migration_test_table) do
+        add :to_be_modified, :string
+        add :to_be_removed, :integer
+      end
+
+      alter table(:migration_test_table) do
+        remove :to_be_removed
+      end
+
+      execute "INSERT INTO migration_test_table (to_be_modified) VALUES ('foo')"
+    end
+
+    def down do
+      drop table(:migration_test_table)
+    end
+  end
+
+  @tag :modify_column
+  test "modify column" do
+    import Ecto.Query, only: [from: 2]
+    assert :ok == up(TestRepo, 20080906120000, AlterColumnMigration, log: false)
+    assert "foo" == TestRepo.one from p in MigrationTestTable, select: p.to_be_modified
+    :ok = down(TestRepo, 20080906120000, AlterColumnMigration, log: false)
+  end
+
+  @tag :remove_column
+  test "remove column" do
+    import Ecto.Query, only: [from: 2]
+    assert :ok == up(TestRepo, 20090906120000, DropColumnMigration, log: false)
+    assert catch_error(TestRepo.one from p in MigrationTestTable, select: p.to_be_removed)
+    unless TestRepo.adapter == Ecto.Adapters.Postgres do
+      :ok = down(TestRepo, 20090906120000, DropColumnMigration, log: false)
+    end
```

Comment:
Yes. You will need to write the tests to create/change/delete tables that are not used anywhere else (name them migration_1, migration_2 and so on) and not use Ecto.Integration.Case in them as MuSQL autocommits. :(